Automattic/wp-calypso
# Components: Refactor Card to TypeScript

#### Changes proposed in this Pull Request

* Migrates `Card` to TypeScript.
* In a future PR `Card` could be converted to a function component and use `React.memo` instead of `PureComponent` but I felt that was out of scope of this PR/wanted to maintain some simplicity here.

#### Testing instructions

* Unit tests and storybook pass
* Ensure `reader-post-card` continues to work (it uses the `tagName` prop for which the logic was slightly changed in this PR).
* In fact, just search `card` in `/devdocs/blocks` and ensure all examples continue to function.
